pkg/datapath/loader/base.go

```diff
@@ -239,14 +240,28 @@ func (l *Loader) reinitializeXDPLocked(ctx context.Context, extraCArgs []string)
 	if option.Config.XDPMode == option.XDPModeDisabled {
 		return nil
 	}
+
+	ignored_xdp_device_regex := regexp.MustCompile(option.Config.XDPIgnoreDeviceNameRegex)
```

Comment:
My initial comment below read:
> call `regexp.MustCompile` and assign to an (unexported) package-scoped variable.

`ignored_xdp_device_regex` is a local variable, not a package-scoped variable. `MustCompile` will still be called on every call to `reinitializeXDPLocked`. What I mean is `var rgxIgnoredXDPDevices = regexp.MustCompile(option.Config.XDPIgnoreDeviceNameRegex)` at the top of the file.

Reply:
This didn't end up working correctly. When the regex is compiled as part of assigning it to a var it does not look like `option.Config.XDPIgnoreDeviceNameRegex` has been initialized yet, so the compiled regex is effectively an empty string. The consequence of this is that all devices match the regex and none get configured for XDP.

I'm not sure what a path forward would be since I have much less experience in the code base than you all. Any guidance here would be _very_ appreciated!

Reply:
Fair enough, this function isn't called often enough for the alternatives to be worth it (e.g. putting it behind a sync.Once). Please do adjust the variable name, though. I recommend giving the document at https://go.dev/doc/effective_go#mixed-caps a read just in case!